File: src/models/Session.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Session = sequelize.define(
  'Session',
  {
    id: { type: DataTypes.BIGINT, primaryKey: true, autoIncrement: true },
    uuid: { type: DataTypes.UUID, defaultValue: DataTypes.UUIDV4, unique: true },
    course_id: { type: DataTypes.BIGINT, allowNull: true },
    teacher_id: { type: DataTypes.BIGINT, allowNull: false },
    subject_id: { type: DataTypes.BIGINT, allowNull: false },
    title: { type: DataTypes.STRING(200), allowNull: false },
    description: { type: DataTypes.TEXT, allowNull: true },
    scheduled_at: { type: DataTypes.DATE, allowNull: false },
    duration_min: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 60 },
    status: {
      type: DataTypes.ENUM('scheduled', 'live', 'ended', 'cancelled'),
      allowNull: false,
      defaultValue: 'scheduled',
    },
    is_private: { type: DataTypes.BOOLEAN, allowNull: false, defaultValue: false },
    price: { type: DataTypes.DECIMAL(12, 2), allowNull: false, defaultValue: 0 },
    room_name: { type: DataTypes.STRING(120), unique: true, allowNull: false },
    recording_enabled: { type: DataTypes.BOOLEAN, allowNull: false, defaultValue: true },
    started_at: { type: DataTypes.DATE, allowNull: true },
    ended_at: { type: DataTypes.DATE, allowNull: true },
  },
  {
    tableName: 'sessions',
    indexes: [
      { fields: ['teacher_id'] },
      { fields: ['subject_id'] },
      { fields: ['status'] },
      { fields: ['scheduled_at'] },
    ],
  },
);

module.exports = Session;

File: mobile/lib/models/session.dart
class Session {
  final int id;
  final String title;
  final String? description;
  final String status;
  final String roomName;
  final double price;
  final bool isPrivate;
  final bool recordingEnabled;
  final String? subjectName;
  final String? teacherName;
  final String? teacherAvatar;
  final int? enrollmentCount;
  final String? scheduledAt;
  final String? startedAt;
  final String? endedAt;
  final int durationMin;

  Session({
    required this.id,
    required this.title,
    this.description,
    this.status = 'scheduled',
    required this.roomName,
    this.price = 0,
    this.isPrivate = false,
    this.recordingEnabled = true,
    this.subjectName,
    this.teacherName,
    this.teacherAvatar,
    this.enrollmentCount,
    this.scheduledAt,
    this.startedAt,
    this.endedAt,
    this.durationMin = 60,
  });

  factory Session.fromJson(Map<String, dynamic> json) => Session(
    id: json['id'] ?? 0,
    title: json['title'] ?? '',
    description: json['description'],
    status: json['status'] ?? 'scheduled',
    roomName: json['room_name'] ?? '',
    price: (json['price'] ?? 0).toDouble(),
    isPrivate: json['is_private'] ?? false,
    recordingEnabled: json['recording_enabled'] ?? true,
    subjectName: json['subject']?['name'],
    teacherName: json['teacher']?['user']?['name'],
    teacherAvatar: json['teacher']?['user']?['avatar'],
    enrollmentCount: json['enrollment_count'],
    scheduledAt: json['scheduled_at'],
    startedAt: json['started_at'],
    endedAt: json['ended_at'],
    durationMin: json['duration_min'] ?? 60,
  );

  bool get isLive => status == 'live';
  bool get isScheduled => status == 'scheduled';
  bool get isEnded => status == 'ended';
  bool get isCancelled => status == 'cancelled';

  String get statusLabel => switch (status) {
    'live' => 'مباشر',
    'scheduled' => 'مجدول',
    'ended' => 'منتهية',
    'cancelled' => 'ملغية',
    _ => status,
  };
}
